common/include/pcl/experimental/executor/omp_executor.hpp

```diff
@@ -0,0 +1,104 @@
+/*
+ * SPDX-License-Identifier: BSD-3-Clause
+ *
+ *  Point Cloud Library (PCL) - www.pointclouds.org
+ *  Copyright (c) 2020-, Open Perception, Inc.
+ *  Author: Shrijit Singh <shrijitsingh99@gmail.com>
+ *
+ */
+
+#pragma once
+
+#ifdef _OPENMP
+  #include <omp.h>
+#endif
+
+#include <pcl/experimental/executor/property.h>
+#include <pcl/experimental/executor/type_trait.h>
+
+namespace executor {
+
+template <typename Blocking, typename ProtoAllocator>
+struct omp_executor;
+
+#ifdef _OPENMP
+template <>
+struct is_executor_available<omp_executor> : std::true_type {};
+#endif
+
+template <typename Blocking = blocking_t::always_t,
+          typename ProtoAllocator = std::allocator<void>>
+struct omp_executor {
+  using shape_type = std::size_t;
+
+  using index_type = struct {
+    shape_type max;
+    int idx;
+  };
+
+  std::size_t max_threads = 0;
+
+  omp_executor(): omp_executor(0) {}
+
+  omp_executor(std::size_t max_threads): max_threads(max_threads) {
+#ifdef _OPENMP
+    set_max_threads(max_threads);
+#endif
+  }
+
+  bool set_max_threads(std::size_t max_threads) {
+#ifdef _OPENMP
+    if (!max_threads) this->max_threads = omp_get_max_threads();
+    else this->max_threads = max_threads;
+    return true;
+#endif
+    return false;
+  }
+
+  template <typename Executor, InstanceOf<Executor, omp_executor> = 0>
+  friend bool operator==(const omp_executor& lhs,
+                         const Executor& rhs) noexcept {
+    pcl::utils::ignore(lhs, rhs);
+    return std::is_same<omp_executor, Executor>::value;
+    pcl::utils::ignore(lhs, rhs);
+  }
+
+  template <typename Executor, InstanceOf<Executor, omp_executor> = 0>
+  friend bool operator!=(const omp_executor& lhs,
+                         const Executor& rhs) noexcept {
+    return !operator==(lhs, rhs);
+  }
+
+  template <typename F>
+  void execute(F&& f) const {
+    static_assert(is_executor_available_v<omp_executor>, "OpenMP executor unavailable");
+    f();
+  }
+
+  template <typename F>
+  void bulk_execute(F&& f, const shape_type& n) const {
+    static_assert(is_executor_available_v<omp_executor>, "OpenMP executor unavailable");
+    pcl::utils::ignore(f, n);
```

Comment:
Yeah, but Clang complains about this. It will throw a static assert failure only if it is called, the compiler still evaluate the entire method, so it will detect unused variables.